### PR TITLE
- Payments screen has been designed. 

### PR DESCRIPTION
**Pull request by yasirinitialcode**
## Related Issue
- Payments screen design. 

ISSUE LINK: https://app.asana.com/0/1203144541549209/1204097456360678

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?
Tested with Redmi Note 7 and Samsung Galaxy M12

## Screenshots or Sample Output
<p float="left">
  
</p>

### CODE REVIEW

**Comment by yasirinitialcode** [comment]
I worked on main because i had already started when we talked about it, i'll create develop after merging this PR.

**Comment by yasirinitialcode** [comment]
One PR only includes resources.

File: app/src/main/java/com/netmera/netmerafintech/data/model/Favorites.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.netmera.netmerafintech.data.model
+
+data class Favorites (
+    var favoriteId: Int? = null,
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/data/model/Favorites.kt:8]: A.O.

File: app/src/main/java/com/netmera/netmerafintech/data/model/Transaction.kt [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
-import android.graphics.Color
 import com.netmera.netmerafintech.R
 
 data class Transaction (
     var transactionId: Int? = null,
     var name: String? = "",
     val price: String? = "",
     var icon: Int = R.drawable.shopping_bag,
-    var iconContainerColor: Int = Color.rgb(0, 120, 255),
+    var iconContainerColor: Int = R.color.blue,
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/data/model/Transaction.kt:11]: A.O.

File: app/src/main/java/com/netmera/netmerafintech/ui/decorators/DividerItemDecorator.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.netmera.netmerafintech.ui.decorators
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ui/decorators/DividerItemDecorator.kt:5]: /*
 * Copyright (c) 2023 Netmera.
 */

Please also apply this copyright info to all of the other classes.

File: app/src/main/java/com/netmera/netmerafintech/ui/home/TransactionsAdapter.kt [hunk 4]
```diff
@@ -1,30 +1,33 @@
...
-import com.netmera.netmerafintech.data.Transaction
+import com.netmera.netmerafintech.data.model.Transaction
 import com.netmera.netmerafintech.databinding.ItemTransactionBinding
 
 
 class TransactionsAdapter(
     private val transactions: List<Transaction>,
-    private val onTransactionItemClick: (transaction: Transaction) -> Unit
+    private val onTransactionItemClick: (transaction: Transaction) -> Unit,
+    private val context: Context,
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ui/home/TransactionsAdapter.kt:19]: A.O.

File: app/src/main/java/com/netmera/netmerafintech/ui/payments/ContactRecyclerAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            binding.apply {
+                cardBackground.setCardBackgroundColor(ContextCompat.getColor(context, item.containerColor))
+                name.text = item.name
+                letter.text = item.name?.substring(0, 1)
+                root.setOnClickListener { onContactItemClick(item) }
+            }
+        }
+    }
+
+    object ContactsComparator : DiffUtil.ItemCallback<Contact>() {
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ui/payments/ContactRecyclerAdapter.kt:37]: Please add comments

File: app/src/main/java/com/netmera/netmerafintech/ui/payments/PayAgainRecyclerAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                    imageBackground.visibility = View.VISIBLE
+                    letter.text = item.name?.substring(0, 1)
+                }
+                name.text = item.name
+                root.setOnClickListener { onFavoriteItemClick(item) }
+            }
+        }
+    }
+
+    object FavoritesComparator : DiffUtil.ItemCallback<Favorites>() {
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ui/payments/PayAgainRecyclerAdapter.kt:40]: Comments are missing

File: app/src/main/java/com/netmera/netmerafintech/ui/payments/PaymentsFragment.kt [hunk 7]
```diff
@@ -1,6 +1,58 @@
...
+import com.netmera.netmerafintech.ui.decorators.DividerItemDecorator
 
 class PaymentsFragment: Fragment() {
+    private lateinit var binding: FragmentPaymentsBinding
+    private lateinit var viewModel: PaymentsViewModel
+
+    private val onFavoritesClick = { favorite: Favorites ->
+//        FavoriteDetailActivity.open(requireActivity(), favorite) TODO open it after favorite detail is created
+    }
+    private val onContactsClick = { contact: Contact ->
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ui/payments/PaymentsFragment.kt:24]: A.O.

File: app/src/main/java/com/netmera/netmerafintech/ui/payments/PaymentsViewModel.kt [hunk 8]
```diff
@@ -1,6 +1,24 @@
...
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.netmera.netmerafintech.data.model.Contact
+import com.netmera.netmerafintech.data.model.Favorites
+import com.netmera.netmerafintech.data.StaticData
 
 class PaymentsViewModel: ViewModel() {
+    private val _favoritesList = MutableLiveData<List<Favorites>>()
+    private val _contactsList = MutableLiveData<List<Contact>>()
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ui/payments/PaymentsViewModel.kt:15]: A.O.

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1,8 +1,12 @@
...
+    <string name="all_accounts">All Accounts</string>
     <string name="analytics">Analytics</string>
     <string name="app_name">NetmeraFintech</string>
+    <string name="ask_money">Ask someone to send you money</string>
     <string name="balance">$5,432.43</string>
+    <string name="by_bank_transfer_or_link">By bank transfer or link</string>
     <string name="card_last_four_digit">•••• 2183</string>
+    <string name="contacts">Contacts</string>
     <string name="current_account_lowercase">Current Account</string>
     <string name="current_account_uppercase">CURRENT ACCOUNT</string>
```
Comment (bsoykal) [inline app/src/main/res/values/strings.xml]: You can also set this programmatically.

**Review by bsoykal — CHANGES_REQUESTED** [review]
Please apply the visual improvements I mentioned.

File: app/src/main/java/com/netmera/netmerafintech/data/Enums.kt [hunk 10]
```diff
@@ -0,0 +1,7 @@
+package com.netmera.netmerafintech.data
+
+enum class CardType(val value: Int) {
+    CARD_BLACK(0),
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/data/Enums.kt]: Don't need to add card anymore. You can simply name it as `Black`, `Blue` and `Yellow`

File: app/src/main/res/values/strings.xml (hunk 9 above)
Comment (ghost) [reply app/src/main/res/values/strings.xml]: Yes, that would be better

**Review by ghost — CHANGES_REQUESTED** [review]
Still needs small changes

**Review by ghost — APPROVED** [review]

**Review by bsoykal — APPROVED** [review]
Very good, thank you.

**Pull request merged**